src/lib/Syntax.hs

```diff
@@ -1510,6 +1529,30 @@ pattern NothingAtom ty = DataCon MaybeDataDef [ty] 0 []
 pattern JustAtom :: Type -> Atom -> Atom
 pattern JustAtom ty x = DataCon MaybeDataDef [ty] 1 [x]
 
+pattern NestOne :: a -> Nest a
+pattern NestOne x = Nest x Empty
+
+pattern BinderAnn :: a -> BinderP a
+pattern BinderAnn x <- ((\case Ignore   ann  -> ann
+                               Bind (_:>ann) -> ann) -> x)
+  where BinderAnn x = Ignore x
+
+pattern NewTypeCon :: Name -> Type -> [DataConDef]
+pattern NewTypeCon con ty <- [DataConDef con (NestOne (BinderAnn ty))]
+  where NewTypeCon con ty =  [DataConDef con (NestOne (Ignore    ty))]
+
+pattern ClassDictDef :: Name
+                     -> LabeledItems Type -> LabeledItems Type -> [DataConDef]
+pattern ClassDictDef conName superclasses methods =
+  [DataConDef conName
+     (Nest (Ignore (RecordTy (NoExt superclasses)))
+     (Nest (Ignore (RecordTy (NoExt methods))) Empty))]
```

Comment:
Shouldn't we use `BinderAnn` here, so that it becomes insensitive to whether the names are used or not? Or should we just maintain an invariant that we don't use any names in class definitions? But if we go with the latter case, why do we ever use `BinderAnn` -- let's just keep our conventions!

Reply:
Changed to `BinderAnn`